### PR TITLE
ci: add script to extract pool information

### PR DESCRIPTION
**Pull request by kerber0x**
## Description and Motivation

This script is a utility to extract the pool information from the pool factory.

## Related Issues

---
## Checklist:

- [ ] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-core/blob/main/CONTRIBUTING.md).
- [ ] My pull request has a sound title and description (not something vague like `Update index.md`)
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation.
- [ ] The code is formatted properly `cargo fmt --all --`.
- [ ] Clippy doesn't report any issues `cargo clippy -- -D warnings`.
- [ ] I have regenerated the schemas if needed `cargo schema`.

### CODE REVIEW

File: scripts/deployment/extract_pools.sh [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    elif [[ "$asset" == factory/* ]]; then
+      processed_asset=$(basename "$asset")
+    elif [[ "$asset" == $chain* ]]; then
+      local query='{"token_info":{}}'
+      local symbol=$($BINARY query wasm contract-state smart $asset "$query" --node $RPC --output json | jq -r '.data.symbol')
+      processed_asset=$symbol
+    fi
+    processed_assets+=("$processed_asset")
+  done
```
Comment (0xFable) [inline scripts/deployment/extract_pools.sh:43]: Pls comment this. Kinda hard to follow with all the if else branching.

**Review by 0xFable — APPROVED** [review]
This looks so handy man! Great work. One comment around readability but functionally it looks good.

File: scripts/deployment/extract_pools.sh (hunk 1 above)
Comment (kerber0x) [reply scripts/deployment/extract_pools.sh:43]: Sure, will do
